tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/source_files/source_files_container_test.ts

```diff
@@ -0,0 +1,133 @@
+import {By} from '@angular/platform-browser';
+
+import {Store} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {DebuggerComponent} from '../../debugger_component';
+import {DebuggerContainer} from '../../debugger_container';
+import {DataLoadState, State} from '../../store/debugger_types';
+import {createDebuggerState, createState} from '../../testing';
+import {AlertsModule} from '../alerts/alerts_module';
+import {ExecutionDataModule} from '../execution_data/execution_data_module';
+import {InactiveModule} from '../inactive/inactive_module';
+import {setUpMonacoFakes, tearDownMonacoFakes} from '../source_code/testing';
+import {StackTraceModule} from '../stack_trace/stack_trace_module';
+import {
+  getFocusedSourceFileContent,
+  getFocusedSourceLineSpec,
+} from '../../store';
+import {TimelineModule} from '../timeline/timeline_module';
+import {SourceFilesContainer} from './source_files_container';
+import {SourceFilesModule} from './source_files_module';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Source Files Container', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+
+  beforeEach(async () => {
+    setUpMonacoFakes();
+    await TestBed.configureTestingModule({
+      declarations: [DebuggerComponent, DebuggerContainer],
+      imports: [
+        AlertsModule,
+        CommonModule,
+        ExecutionDataModule,
+        InactiveModule,
+        SourceFilesModule,
+        StackTraceModule,
+        TimelineModule,
+      ],
+      providers: [provideMockStore(), DebuggerContainer],
+    }).compileComponents();
+    store = TestBed.get(Store);
+    dispatchSpy = spyOn(store, 'dispatch');
+  });
+
+  afterEach(() => {
+    tearDownMonacoFakes();
+  });
+
+  it('renders no file selected when no source line is focused on', () => {
+    const fixture = TestBed.createComponent(SourceFilesContainer);
+    store.setState(createState(createDebuggerState()));
+    fixture.detectChanges();
+
+    const noFileSelectedElement = fixture.debugElement.query(
+      By.css('.no-file-selected')
+    );
+    expect(noFileSelectedElement.nativeElement.innerText).toBe(
+      '(No file selected)'
+    );
+
+    const sourceCodeElements = fixture.debugElement.queryAll(
+      By.css('source-code-component')
+    );
+    expect(sourceCodeElements.length).toBe(0);
+  });
+
+  it('renders file path and editor when a file is focused on', async () => {
+    const fixture = TestBed.createComponent(SourceFilesContainer);
+    store.overrideSelector(getFocusedSourceFileContent, {
+      loadState: DataLoadState.LOADED,
+      lines: ['import tensorflow as tf', '', 'print("hello, world")'],
+    });
+    store.overrideSelector(getFocusedSourceLineSpec, {
+      host_name: 'localhost',
+      file_path: '/home/user/main.py',
+      lineno: 3,
+    });
+    fixture.detectChanges();
+    await fixture.whenStable();
```

Comment:
I believe you no longer need this since you aren't rendering monaco component? (source_code_component)

Reply:
Feel free to use ` import {NO_ERRORS_SCHEMA} from '@angular/core';`, too.